strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/storage/validation/version/PypiCanonicalVersionValidator.java

```diff
@@ -0,0 +1,126 @@
+        implements PypiVersionValidator
+{
+
+    public static final String EPOCH_FORMAT = "(\\d+!)";
+    public static final String FINAL_RELEASE_FORMAT = "(\\d+(\\.\\d+)*)";
+    public static final String PRE_RELEASE_FORMAT = "((a|b|c|rc)\\d+)";
+    public static final String POST_RELEASE_FORMAT = "(\\.post\\d+)";
+    public static final String DEVELOPMENTAL_RELEASE_FORMAT = "(\\.dev\\d+)";
+
+    //must start and end with ASCII character, can have periods in between
+    public static final String LOCAL_VERSION_IDENTIFIER_FORMAT = "(\\+[a-zA-Z0-9](\\.*[a-zA-Z0-9])*)";
+    public static final String CANONICAL_VERSION_FORMAT = getCanonicalVersionFormat();
+
+
+    public static final Pattern PRE_RELEASE_PATTERN = Pattern.compile(PypiCanonicalVersionValidator.PRE_RELEASE_FORMAT);
+    public static final Pattern POST_RELEASE_PATTERN = Pattern.compile(
+            PypiCanonicalVersionValidator.POST_RELEASE_FORMAT);
+    public static final Pattern DEVELOPMENTAL_RELEASE_PATTERN = Pattern.compile(
+            PypiCanonicalVersionValidator.DEVELOPMENTAL_RELEASE_FORMAT);
+    public static final Pattern LOCAL_VERSION_IDENTIFIER_PATTERN = Pattern.compile(
+            PypiCanonicalVersionValidator.LOCAL_VERSION_IDENTIFIER_FORMAT);
+    public static final Pattern CANONICAL_PATTERN = Pattern.compile(
+            PypiCanonicalVersionValidator.CANONICAL_VERSION_FORMAT);
+
+    @Inject
+    private ArtifactCoordinatesValidatorRegistry artifactCoordinatesValidatorRegistry;
+    private static final Logger logger = LoggerFactory.getLogger(ArtifactCoordinatesValidatorRegistry.class);
+
+    public static final String ALIAS = "pypi-canonical-version-validator";
+
+    public static final String DESCRIPTION = "PyPi canonical version validator";
+
+    @Override
+    public String getDescription()
+    {
+        return DESCRIPTION;
+    }
+
+    @Override
+    public void validate(Repository repository,
+                         ArtifactCoordinates coordinates)
+            throws VersionValidationException
+    {
+        final Matcher matcher = PypiCanonicalVersionValidator.CANONICAL_PATTERN.matcher(coordinates.getVersion());
+        if (!matcher.matches())
+        {
+            throw new VersionValidationException(String.format("Artifact version [%s] should follow the Canonical " +
+                                                               "Versioning specification (https://www.python.org/dev/peps/pep-0440/).",
+                                                               coordinates.getVersion()));
+        }
+    }
+
+    @Override
+    public String getAlias()
+    {
+        return ALIAS;
+    }
+
+    @PostConstruct
+    @Override
+    public void register()
+    {
+        artifactCoordinatesValidatorRegistry.addProvider(ALIAS, this);
+
+        logger.info("Registered artifact coordinates validator '" + getClass().getCanonicalName() + "'" +
+                    " with alias '" + ALIAS + "'.");
```

Comment:
Please, use built-in parameters with logger methods.
e.g.:  `logger.info("Nothing found in db for A with a : {} and b : {}", a ,b);`